ops/testing.py

```diff
@@ -1253,14 +1261,17 @@ def storage_get(self, storage_name_id, attribute):
                 'ERROR invalid value "{}/{}" for option -s: storage not found'.format(name, index))
 
     def storage_add(self, name: str, count: int = 1):
+        if name.startswith('/'):
+            raise model.ModelError('test storage name cannot start with "/"')
```

Comment:
is it **test storage** specifically, or storage names in general that cannot start with /? If the former: why the discrepancy? If the latter: remove 'test'.

Reply:
I actually think you'd probably have problems if you tried to put a slash anywhere in the storage name since they get munged with paths a lot.  I don't know if this is a hard rule for juju though.

Reply:
In fact, this code https://github.com/canonical/operator/blob/e3bc30bf8491716d33c8bb6061eb63bd8306ca1d/ops/main.py#L154-L157 would clearly be broken if the storage name contained a `/`.  This is worth looking into further...

Reply:
Okay.  Found the answer.  Juju uses this to validate storage names: https://github.com/juju/names/blob/9c6532a528238568c78ebb8a41bbb20340da3f90/storage.go#L17-L20 so no slashes allowed.

Reply:
I updated the error check to just disallow slashes in anywhere in the name.